File: src/main/java/com/ject/studytrip/auth/domain/repository/RefreshTokenRedisRepository.java
package com.ject.studytrip.auth.domain.repository;

public interface RefreshTokenRedisRepository {
    void saveRefreshToken(String memberId, String refreshToken, long refreshTokenExpireTime);

    boolean existsRefreshToken(String refreshToken);

    void deleteRefreshToken(String refreshToken);

    String findMemberIdByRefreshToken(String refreshToken);
}

File: src/main/kotlin/com/ject/studytrip/auth/domain/repository/RefreshTokenRedisRepository.kt
package com.ject.studytrip.auth.domain.repository

interface RefreshTokenRedisRepository {
    fun saveRefreshToken(
        memberId: String,
        refreshToken: String,
        refreshTokenExpireTime: Long,
    )

    fun existsRefreshToken(refreshToken: String?): Boolean

    fun deleteRefreshToken(refreshToken: String?)

    fun findMemberIdByRefreshToken(refreshToken: String?): String?
}
